### PR TITLE
Add --format to depend-info.

### PR DESCRIPTION
**Pull request by BillyONeal**
Add error handling and testing to depend-info switch handling.

Documentation update: https://github.com/microsoft/vcpkg-docs/pull/89

### CODE REVIEW

File: include/vcpkg/base/message-data.inc.h [hunk 1]
```diff
@@ -466,16 +466,31 @@ DECLARE_MESSAGE(CmdAddVersionOptSkipFormatChk, (), "", "Skips the formatting che
...
+    "The alternatives in ``s must not be localized. {value} is what the user specified.",
+    "--format={value} is not a recognized format. --format must be one of `list`, `tree`, `dot`, or `dgml`.")
+DECLARE_MESSAGE(CmdDependInfoShowDepthFormatMismatch,
                 (),
                 "",
-                "Set sort order for the list of dependencies, accepted values are: lexicographical, topological "
-                "(default), x-tree, "
-                "reverse")
+                "--show-depth can only be used with `list` and `tree` formats.")
+DECLARE_MESSAGE(CmdDependInfoXtreeTree, (), "", "--sort=xtree can not be used with formats other than tree")
```
Comment (markle11m) [inline include/vcpkg/base/message-data.inc.h:486]: 'can not' --> 'cannot'

File: locales/messages.json [hunk 2]
```diff
@@ -272,11 +272,17 @@
...
+  "CmdDependInfoFormatHelp": "Choose output format, one of `list`, `tree`, `dot`, or `dgml`.",
+  "_CmdDependInfoFormatHelp.comment": "The alternatives in ``s must not be localized.",
+  "CmdDependInfoFormatInvalid": "--format={value} is not a recognized format. --format must be one of `list`, `tree`, `dot`, or `dgml`.",
+  "_CmdDependInfoFormatInvalid.comment": "The alternatives in ``s must not be localized. {value} is what the user specified.",
+  "CmdDependInfoOptDepth": "Show recursion depth in `list` output.",
+  "CmdDependInfoOptMaxRecurse": "Set max recursion depth. Default is no limit.",
+  "CmdDependInfoOptSort": "Choose sort order for the `list` format, one of `lexicographical`, `topological` (default), `reverse`",
+  "_CmdDependInfoOptSort.comment": "The alternatives in ``s must not be localized, but the localized text can explain what each value means. The value `reverse` means 'reverse-topological'.",
+  "CmdDependInfoShowDepthFormatMismatch": "--show-depth can only be used with `list` and `tree` formats.",
+  "CmdDependInfoXtreeTree": "--sort=xtree can not be used with formats other than tree",
```
Comment (markle11m) [inline locales/messages.json:285]: Same as other comment. Shouldn't this be `--sort=x-tree`

Comment (BillyONeal) [reply locales/messages.json]: Yep, fixed!

File: include/vcpkg/base/message-data.inc.h (hunk 1 above)
Comment (BillyONeal) [reply include/vcpkg/base/message-data.inc.h]: Fixed!

**Review by markle11m — APPROVED** [review]

**Pull request merged**